ts/WoltLabSuite/Core/Ui/Page/Menu/Main.ts

```diff
@@ -314,6 +319,28 @@ export class PageMenuMain implements PageMenuProvider {
     } else {
       this.mainMenu.classList.remove("pageMenuMobileButtonHasContent");
     }
+
+    const menuItems = this.menuItemProvider.getMenuItems(this.mainMenu);
+    menuItems.forEach((menuItem) => this.refreshUnreadBage(menuItem));
+  }
+
+  private refreshUnreadBage(menuItem: MenuItem): void {
+    if (menuItem.identifier !== false) {
+      const counter = this.menuItemBadges.get(menuItem.identifier);
+      if (counter) {
+        if (menuItem.counter === 0) {
+          counter.remove();
+          this.menuItemBadges.delete(menuItem.identifier);
+        } else {
+          const value = parseInt(counter.textContent!, 10);
+          if (value !== menuItem.counter) {
```

Comment:
Is this check necessary (e.g. for performance)? Otherwise I'd remove it (and the line above) to keep the indentation level lower.

Reply:
DOM operations are always somewhat expensive, updating the `textContent` is effectively this operation:
```ts
counter.childNodes[0].remove();
counter.append(document.createTextNode(menuItem.counter.toString()));
```

We can simply this into a one-liner, but it doesn‘t really make much of a difference, does it?

Reply:
> DOM operations are always somewhat expensive,

Understood. Then leave it as it is. A one-liner doesn't bring any benefit here if a conditional still is required.